pkg/utils/etcdutil/health_checker.go

```diff
@@ -152,25 +163,147 @@ func (checker *healthChecker) patrol(ctx context.Context) []string {
 			defer logutil.LogPanic()
 			var (
 				ep     = key.(string)
-				client = value.(*healthyClient)
+				client = value.(*healthyClient).Client
+				start  = time.Now()
 			)
-			if IsHealthy(ctx, client.Client) {
-				hch <- ep
+			healthy := IsHealthy(ctx, client)
+			// If the endpoint is healthy, update its last health time.
+			if healthy {
 				checker.storeClient(ep, &healthyClient{
-					Client:     client.Client,
-					lastHealth: time.Now(),
+					Client:     client,
+					lastHealth: start,
 				})
-				return
+			}
+			// Send the probe result to the channel.
+			probeCh <- healthProbe{
+				ep:      ep,
+				healthy: healthy,
+				took:    time.Since(start),
 			}
```

Comment:
seems like can remove the `healthy` check?
we can only pass `healthy` healthProbe in probech?

Reply:
Addressed.